sentry/src/main/java/io/sentry/protocol/SentryId.java

```diff
@@ -5,48 +5,58 @@
 import io.sentry.JsonSerializable;
 import io.sentry.ObjectReader;
 import io.sentry.ObjectWriter;
+import io.sentry.util.LazyEvaluator;
 import io.sentry.util.StringUtils;
 import java.io.IOException;
 import java.util.UUID;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public final class SentryId implements JsonSerializable {
-  private final @NotNull UUID uuid;
 
   public static final SentryId EMPTY_ID = new SentryId(new UUID(0, 0));
 
+  private final @NotNull LazyEvaluator<UUID> lazyValue;
+
   public SentryId() {
     this((UUID) null);
   }
 
   public SentryId(@Nullable UUID uuid) {
-    if (uuid == null) {
-      uuid = UUID.randomUUID();
+    if (uuid != null) {
+      this.lazyValue = new LazyEvaluator<>(() -> uuid);
+    } else {
+      this.lazyValue = new LazyEvaluator<>(UUID::randomUUID);
     }
-    this.uuid = uuid;
   }
 
   public SentryId(final @NotNull String sentryIdString) {
-    this.uuid = fromStringSentryId(StringUtils.normalizeUUID(sentryIdString));
+    if (sentryIdString.length() != 32 && sentryIdString.length() != 36) {
+      throw new IllegalArgumentException(
+          "String representation of SentryId has either 32 (UUID no dashes) "
+              + "or 36 characters long (completed UUID). Received: "
+              + sentryIdString);
+    }
+    this.lazyValue =
+        new LazyEvaluator<>(() -> fromStringSentryId(StringUtils.normalizeUUID(sentryIdString)));
   }
 
   @Override
   public String toString() {
-    return StringUtils.normalizeUUID(uuid.toString()).replace("-", "");
+    return StringUtils.normalizeUUID(lazyValue.getValue().toString()).replace("-", "");
```

Comment:
This toString is called from multiple places. since uuid wont be changes. it is good to cache that. but also this toString will do unnecessary calculations.
Please consider holding toString value instead of uuid similar to SpanId or also cache the toString result.

Reply:
It is more related to https://github.com/getsentry/sentry-java/issues/3772
But if toString is called right after creating SentryId, it wont help much of delaying this UUID cost.

Reply:
@adinauer -> Please see comment from @kozaxinan

Reply:
Yeah, we should measure impact of this PR. I'm afraid this doesn't have the desired effect due to `toString()`, `hashCode` and `equals` being called early. Caching the value for `toString()` might have a bigger impact, thanks for the suggestion.
Changing how we use IDs (i.e. only generate them when sending) will be a bigger change and I'm not entirely certain it's possible. Offering a configurable ID generation mechanism might be the better approach. We'll discuss internally and update once we know more.

Reply:
@kozaxinan Good point, I'll rework `SentryId` to cache the resulting `String` instead of the `UUID` object.

As mentioned by @adinauer, measuring the performance impact of this change has yet to be done.

Reply:
Nice this will allow more improvement. Will this be ported to version 7?

Reply:
@kozaxinan we're planning to explore multiple options to improve the SDK. We can check which of those make sense to port to v7 when we know which ones we'll keep.